build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/LegacyRestTestBasePlugin.java

```diff
@@ -121,6 +134,27 @@ public void apply(Project project) {
         });
     }
 
+    private void configureCacheability(RestIntegTestTask restIntegTestTask) {
```

Comment:
Having this logic within the constructor of the StandaloneRestIntegTest task breaks configuration cache compatibility. Now StandaloneRestIntegTest task itself is configuration cache compatible and we can use it for new test setup like we do in non legacy plugins